app/controllers/items_controller.rb

```diff
@@ -44,7 +49,7 @@ def set_item
   end
 
   def move_to_index
-    unless @item == current_user.id
+    unless @item.user.id.to_i == current_user.id
```

Comment:
```suggestion
    unless @item.user.id == current_user.id
```
to_iメソッドとは、数字の文字列を数値オブジェクトに変換するメソッドです。
`@item.user.id`では、商品出品者のid（数値）を取得します。
変換しなくとも最初から数値を取得するため、`@item.user.id`に修正しておきましょう。

Reply:
修正しました